frame/auction.ts

```diff
@@ -44,23 +45,22 @@ export async function runAdAuction(
   if (firstAdResult.done) {
     return null;
   }
-  let [winningRenderingUrl, winningPrice] = firstAdResult.value;
-  const renderingUrls = new Set([winningRenderingUrl]);
-  for (const [renderingUrl, price] of ads) {
-    renderingUrls.add(renderingUrl);
-    if (price > winningPrice) {
-      winningRenderingUrl = renderingUrl;
-      winningPrice = price;
+  let winner = firstAdResult.value;
+  const renderingUrls = new Set([winner.renderingUrl]);
+  for (const ad of ads) {
+    renderingUrls.add(ad.renderingUrl);
+    if (ad.metadata.price > winner.metadata.price) {
+      winner = ad;
     }
   }
-  if (trustedScoringSignalsUrl !== null) {
+  if (trustedScoringSignalsUrl !== undefined) {
```

Comment:
nit: just `if (trustedScoringSignalsUrl)` would be fine

Reply:
Would rather not have the empty string be treated differently; it's more predictable if I know it'll just try to be parsed as a URL and fail